src/FormParseNode.cs

```diff
@@ -184,17 +204,26 @@ public IEnumerable<T> GetCollectionOfPrimitiveValues<T>()
         // Parse an ISO8601 duration.http://en.wikipedia.org/wiki/ISO_8601#Durations to a TimeSpan
         return XmlConvert.ToTimeSpan(rawString);
     }
+
     /// <inheritdoc/>
     public Time? GetTimeValue() => DateTime.TryParse(DecodedValue, out var result) ? new Time(result) : null;
+
 #if NET5_0_OR_GREATER
     IEnumerable<T?> IParseNode.GetCollectionOfEnumValues<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)] T>()
 #else
     IEnumerable<T?> IParseNode.GetCollectionOfEnumValues<T>()
 #endif
     {
-        foreach (var v in DecodedValue.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries))
-            yield return GetEnumValueInternal<T>(v);
+        List<T?> enumValues = new List<T?>();
+        string[] values = DecodedValue.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+        foreach (string v in values)
+        {
+            enumValues.Add(GetEnumValueInternal<T>(v));
+        }
+        return enumValues;
```

Comment:
I'm not sure I understand why this change is necessary?

Reply:
+1 ... It's not necessary. `yield return` generates a perfectly fine enumerator instance.

Reply:
There's no LINQ in the method already. It's a generated iterator which is not problematic per se.